src/wx/faudio.cpp

```diff
@@ -194,43 +190,44 @@ class FAudio_Device_Notifier : public IMMNotificationClient {
 // Synchronization Event
 class FAudio_BufferNotify : public FAudioVoiceCallback {
 public:
-    HANDLE hBufferEndEvent;
+    void *hBufferEndEvent;
```

Comment:
Wouldn't that run afoul for it's use later, like if (WaitForSingleObject(notify.hBufferEndEvent, 10000) == WAIT_TIMEOUT), wouldn't it complain about it being declared private? I guess I'm just not really understanding the thought process behind the suggestion.

Reply:
Private members of a class are only accessible by the methods of said class. Since the `WaitForSingleObject()` call is done within the class itself, it will work just fine. At the end of the day, this is just data that is passed to the `WaitForSingleObject()` call. The point of private members is to prevent non-member functions from accidentally accessing them.

Reply:
Oh, nevermind, I just re-read the code and realized it's used by a different object.
I would suggest turning the `WaitForSingleObject()` call into an utilitly method in `FAudio_BufferNotify`, like this:
```c++
// Waits for this object to be notified for 10 seconds. Returns true if this object was signaled.
bool WaitForSignal() {
    return WaitForSingleObject(hBufferEvent, 10000) != WAIT_TIMEOUT;
}
```

It will also make it easier to port to non-Windows down the line.